### PR TITLE
fix(ldap): sql errors in the log on authentication

### PR DESCRIPTION
**Pull request by sc979**
<h1> LDAP SQL errors in the logs </h1>

<h2> Description </h2>

Fixes # (issue): support

<h2> Type of change </h2>

Cleaning code
Fixing : "Undefined index: contact_passwd" in php logs
Fixing: "QUERY : SELECT contact_id FROM contact WHERE contact_ldap_dn IN ()" in sql logs

- [X] Patch fixing an issue (non-breaking change)
- [ ] New functionality (non-breaking change)
- [ ] Breaking change (patch or feature) that might cause side effects breaking part of the Software
- [ ] Updating documentation (missing information, typo...)

<h2> Target serie </h2>

- [X] 18.10.x
- [X] 19.04.x (master)

<h2> How this pull request can be tested ? </h2>

When importing an user from the LDAP, you should find these errors before applying the fix.

<h2> Checklist </h2>

<h5> Community contributors & Centreon team </h5>

- [X] I followed the **coding style guidelines** provided by Centreon
- [X] I have commented my code, especially new **classes**, **functions** or any **legacy code** modified.
- [X] I have **rebased** my development branch on the base branch (master, maintenance).

<h5> Centreon team only </h5>

- [X] I have made sure that the **unit tests** related to the story are successful.
- [ ] I have made sure that **unit tests covers 80%** of the code written for the story.
- [X] I have made sure that **acceptance tests** related to the story are successful (**local and CI**)

### CODE REVIEW

File: www/class/centreonContactgroup.class.php [hunk 1]
```diff
@@ -201,6 +201,85 @@ public function insertLdapGroup($cg_name)
...
         return $row['cg_id'];
     }
+    
+    /**
+     * Synchronize with LDAP groups at config generation
+     *
+     * @return array | array of error messages
+     */
+    public function syncWithLdapConfigGen()
+    {
```
Comment (jdelpierre) [inline www/class/centreonContactgroup.class.php]: For this function, I find the commentary too light.
The code is hard to read, to understand, you have to read line by line.
The name of the variables is not very explicit. 
Would it be possible to divide certain parts by block or function? Or at least comment on several places in the code?

Comment (sc979) [reply www/class/centreonContactgroup.class.php]: I forked a PR deployed on the customer's prod.
But this part of the code is from : 
https://github.com/centreon/centreon/pull/7130/files
As it's still waiting for QA validation, I couldn't do an other way without creating multiple conflicts. And this part of the PR needed to be fixed too.

File: www/class/centreonContactgroup.class.php [hunk 2]
```diff
@@ -174,33 +176,109 @@ public function insertLdapGroup($cg_name)
...
+
+            if ($contact !== '') {
+                $queryContact = "SELECT contact_id FROM contact WHERE contact_ldap_dn IN (" . $contact . ")";
+                try {
+                    $resContact = $this->db->query($queryContact);
+                } catch (\PDOException $e) {
+                    $msg[] = "Error in getting contact id form members.";
+                    continue;
+                }
+                while ($rowContact = $resContact->fetch()) {
```
Comment (jdelpierre) [inline www/class/centreonContactgroup.class.php]: While in the Try

File: www/class/centreonContactgroup.class.php [hunk 3]
```diff
@@ -174,33 +176,109 @@ public function insertLdapGroup($cg_name)
...
+                $ldapConn = new CentreonLDAP($this->db, null, $cgRow['ar_id']);
+                $connectionResult = $ldapConn->connect();
+                if ($connectionResult == false) {
+                    $ldapServerConnError[$cgRow['ar_id']] = 1;
+                    $msg[] = "Unable to connect to LDAP server.";
+                    continue;
+                }
+            }
+
+            $members = $ldapConn->listUserForGroup($cgRow['cg_ldap_dn']);
```
Comment (jdelpierre) [inline www/class/centreonContactgroup.class.php]: $members move to line 253

File: www/class/centreonContactgroup.class.php [hunk 4]
```diff
@@ -174,33 +176,109 @@ public function insertLdapGroup($cg_name)
...
+            if ($ar_id != $cgRow['ar_id']) {
+                $ar_id = $cgRow['ar_id'];
+                if (!is_null($ldapConn)) {
+                    $ldapConn->close();
+                }
+                $ldapConn = new CentreonLDAP($this->db, null, $cgRow['ar_id']);
+                $connectionResult = $ldapConn->connect();
+                if ($connectionResult == false) {
+                    $ldapServerConnError[$cgRow['ar_id']] = 1;
+                    $msg[] = "Unable to connect to LDAP server.";
```
Comment (jdelpierre) [inline www/class/centreonContactgroup.class.php]: it is possible to enrich the log with at least the ID concerned

File: www/class/centreonContactgroup.class.php [hunk 5]
```diff
@@ -174,33 +176,109 @@ public function insertLdapGroup($cg_name)
...
+                $contact .= $this->db->quote($member) . ',';
+            }
+            $contact = rtrim($contact, ",");
+
+            if ($contact !== '') {
+                $queryContact = "SELECT contact_id FROM contact WHERE contact_ldap_dn IN (" . $contact . ")";
+                try {
+                    $resContact = $this->db->query($queryContact);
+                } catch (\PDOException $e) {
+                    $msg[] = "Error in getting contact id form members.";
```
Comment (jdelpierre) [inline www/class/centreonContactgroup.class.php]: it is possible to enrich the log with at least the ID concerned.

File: www/class/centreonContactgroup.class.php (hunk 4 above)
Comment (callapa) [inline www/class/centreonContactgroup.class.php]: Please add more information about the LDAP server

File: www/class/centreonContactgroup.class.php [hunk 6]
```diff
@@ -174,33 +176,109 @@ public function insertLdapGroup($cg_name)
...
+                    continue;
+                }
+                while ($rowContact = $resContact->fetch()) {
+                    $queryAddRelation = "INSERT INTO contactgroup_contact_relation " .
+                        "(contactgroup_cg_id, contact_contact_id) " .
+                        "VALUES (" . $cgRow['cg_id'] . ", " . $rowContact['contact_id'] . ")";
+                    try {
+                        $this->db->query($queryAddRelation);
+                    } catch (\PDOException $e) {
+                        $msg[] = "Error insert relation between contactgroup " . $cgRow['cg_id'] .
```
Comment (callapa) [inline www/class/centreonContactgroup.class.php]: Please to replace ids by contractgroup name and contact name.

**Pull request merged**